packages/experimental/TabList/src/Tab/Tab.tsx

```diff
@@ -28,74 +33,84 @@ const tabLookup = (layer: string, state: TabState, props: TabProps, tablistConte
+  [tokens, cache] = applyTokenLayers(tokens, tabStates, cache, (layer) => tabLookup(layer, tab.state, tab.props, tablist));
+
+  // Get styling props for each Tab slot
+  const slotProps = useTabSlotProps(tab.props, tokens, theme, tablist);
 
-      const { icon, tabKey, ...mergedProps } = mergeProps(tab.props, final, {
-        accessibilityLabel: tab.props.accessibilityLabel || final.accessibilityLabel || label,
-      });
+  const onTabLayout = useTabAnimation(props, tablist, tokens);
 
-      if (__DEV__ && !hasChildren && !icon) {
-        console.warn('A Tab component must render content. Children, an icon, or both should be passed in.');
+  const RootSlot = useSlot<PressablePropsExtended>(Pressable, getRootProps(slotProps.root, onTabLayout));
+  const StackSlot = useSlot<ViewProps>(View, slotProps.stack as ViewProps);
+  const IndicatorContainerSlot = useSlot<ViewProps>(View, slotProps.indicatorContainer as ViewProps);
+  const IndicatorSlot = useSlot<ViewProps>(View, slotProps.indicator as ViewProps);
+  const ContentContainerSlot = useSlot<ViewProps>(View, slotProps.contentContainer as ViewProps);
+  const ContentSlot = useSlot<TextProps>(Text, slotProps.content);
+  const IconSlot = useSlot<IconProps>(Icon, slotProps.icon);
+
+  return (final: TabProps, ...children: React.ReactNode[]) => {
+    if (!tab.state) {
+      return null;
+    }
+
+    // Get label for Tab to use if there's no accessibilityLabel prop passed in.
+    let label = '';
+    let hasChildren = false;
+    React.Children.forEach(children, (child) => {
+      if (child !== null) {
+        hasChildren = true;
+        if (typeof child === 'string') {
+          label = child;
+        }
       }
+    });
+
+    const { icon, tabKey, ...mergedProps } = mergeProps(tab.props, final, {
+      accessibilityLabel: tab.props.accessibilityLabel || final.accessibilityLabel || label,
+    });
+
+    if (__DEV__ && !hasChildren && !icon) {
+      console.warn('A Tab component must render content. Children, an icon, or both should be passed in.');
+    }
 
-      return (
-        <Slots.root {...mergedProps}>
-          <Slots.stack>
-            {icon && <Slots.icon {...icon} />}
-            {hasChildren && (
-              <Slots.contentContainer>
-                {React.Children.map(children, (child, i) =>
-                  typeof child === 'string' ? (
-                    <Slots.content accessible={false} key={i}>
-                      {child}
-                    </Slots.content>
-                  ) : (
-                    child
-                  ),
-                )}
-              </Slots.contentContainer>
-            )}
-          </Slots.stack>
-          <Slots.indicator />
-        </Slots.root>
-      );
-    };
-  },
-});
+    return (
+      <RootSlot {...mergedProps}>
+        <StackSlot>
+          {icon && <IconSlot {...icon} />}
+          {hasChildren && (
+            <ContentContainerSlot>
+              {React.Children.map(children, (child, i) =>
```

Comment:
this is a bit of a drive-by comment since it's the same as before, but what scenario was this used for? I'm intrigued at why we can't natively pass in {children} and the effect on varying combinations (more than one string child or mixed types)

Reply:
Talked about offline, this is used to pass in both string and non-string children and have them render correctly.